src/globus_sdk/local_endpoint/server/endpoint.py

```diff
@@ -0,0 +1,84 @@
+import json
+import pathlib
+import typing as t
+
+
+class LocalGlobusConnectServer:
+    r"""
+    A LocalGlobusConnectServer object represents the available SDK methods
+    for inspecting and controlling a running Globus Connect Server
+    installation.
+
+    These objects do *not* inherit from BaseClient and do not provide methods
+    for interacting with any Globus Service APIs.
+
+    :param info_path: The path to the info file used to inspect the local system
+    :type info_path: str, optional
+    """
+
+    def __init__(
+        self,
+        *,
+        info_path: t.Union[
+            str, pathlib.Path
+        ] = "/var/lib/globus-connect-server/info.json"
+    ) -> None:
+        self.info_path = pathlib.Path(info_path)
+        self._loaded_info: t.Optional[t.Dict[str, t.Any]] = None
+
+    @property
+    def info_dict(self) -> t.Optional[t.Dict[str, t.Any]]:
+        """
+        The info.json data for the local Globus Connect Server, as a dict.
+
+        If the info.json file is not present or cannot be parsed, the data is None.
+        This indicates that there is no local Globus Connect Server node, or if there is
+        one, it cannot be used or examined by the SDK. For example, containerized
+        applications using the SDK may not be able to interact with Globus Connect
+        Server on the container host.
+
+        :type: dict
+        """
+        if self._loaded_info is None:
+            if self.info_path.is_file():
+                with open(self.info_path, encoding="utf-8") as fp:
+                    try:
+                        parsed_data = json.load(fp)
+                    except json.JSONDecodeError:
```

Comment:
`UnicodeDecodeError` needs to be caught as well.

```suggestion
                    except (UnicodeDecodeError, json.JSONDecodeError):
```

Reply:
I always assumed that the decoder would raise a `JSONDecodeError` on non-unicode data. I'm willing to bet that I have this handling missing from many bits of code I've written over the years.

I'll put together a test case and get this added.